### DISCUSSION TITLE
json_query with jsonpath::result_options::path | jsonpath::result_options::value ?

### DISCUSSION THREAD

**Discussion by harsszegi** [danielaparker/jsoncons#434] [General] [answered]
Hello,

is it possible to do a json_query so that the returned result would be a list of pairs with the path of the node and the value of it?
[Without possibly doing the query twice]
Thanks,

**Comment by jsonconsorg** [comment, marked as answer]
Yes. Have a look at the json_query example that takes a callback with two parameters, the path and the value. You can collect all the path value pairs from the callback.

**Reply by harsszegi** [reply]
Awesome, thanks a lot!!
